ir/memory.h

```diff
@@ -157,7 +217,15 @@ class Memory {
   static Memory mkIf(const smt::expr &cond, const Memory &then,
                      const Memory &els);
 
+  // Returns the number of bits needed to represent a pointer offset.
   unsigned bitsOffset() const { return bits_for_offset; }
+  // Returns the number of bits needed to represent a block id.
+  unsigned bitsBid() const { return bits_for_local_bid +
+                                    bits_for_nonlocal_bid; }
+  // Returns the number of bits needed to represent a byte.
+  unsigned bitsByte() const { return 1 + 1 + bitsBid() + bitsOffset() + 3; }
+  // Returns the number of bits needed to represent the size of memory.
+  unsigned bitsMemorySize() const { return bits_size_t; }
```

Comment:
Memory size is a strange name. Either use ptrSize() or something.
Number of bits of a pointer is not the same as the log(memory-size), since not all archs are byte addressable.